rust/stracciatella_c_api/src/c/path.rs

```diff
@@ -0,0 +1,53 @@
+//! This module contains the C interface for [`std::path`], excluding filesystem interactions.
+//!
+//! [`std::path`]: https://doc.rust-lang.org/std/path/index.html
+
+use std::ptr;
+
+use crate::c::common::*;
+
+/// Gets the extension of the path.
+/// Returns null if there is no extension.
+#[no_mangle]
+pub extern "C" fn Path_extension(path: *const c_char) -> *mut c_char {
```

Comment:
We could also keep `Path` as a type for C++ and convert to `c_char` whenever needed in C++ code. This way we would represent paths as their original OS-dependent representation. I know its not done in other parts of the code, but might be a thought for the future...

Reply:
I plan to rethink paths when the non-boost filesystem stuff is moved to rust.